modules/reindex/src/main/java/org/opensearch/index/reindex/remote/RemoteVersion.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.reindex.remote;
+
+import java.util.Locale;
+import java.util.Objects;
+
+/**
+ * Represents a version of a remote Elasticsearch or OpenSearch cluster for reindexing purposes.
+ * This class provides backward compatibility support for communicating with older Elasticsearch versions
+ * without relying on the global Version class.
+ */
+public final class RemoteVersion implements Comparable<RemoteVersion> {
+
+    private final int major;
+    private final int minor;
+    private final int revision;
+    private final String distribution;
+
+    // Common version constants for backward compatibility
+    public static final RemoteVersion V_0_20_5 = new RemoteVersion(0, 20, 5, null);
+    public static final RemoteVersion V_0_90_13 = new RemoteVersion(0, 90, 13, null);
+    public static final RemoteVersion V_1_0_0 = new RemoteVersion(1, 0, 0, null);
+    public static final RemoteVersion V_1_7_5 = new RemoteVersion(1, 7, 5, null);
+    public static final RemoteVersion V_2_0_0 = new RemoteVersion(2, 0, 0, null);
+    public static final RemoteVersion V_2_1_0 = new RemoteVersion(2, 1, 0, null);
+    public static final RemoteVersion V_2_3_3 = new RemoteVersion(2, 3, 3, null);
+    public static final RemoteVersion V_5_0_0 = new RemoteVersion(5, 0, 0, null);
+    public static final RemoteVersion V_6_0_0 = new RemoteVersion(6, 0, 0, null);
+    public static final RemoteVersion V_6_3_0 = new RemoteVersion(6, 3, 0, null);
+    public static final RemoteVersion V_7_0_0 = new RemoteVersion(7, 0, 0, null);
+
+    // OpenSearch versions
+    public static final RemoteVersion OPENSEARCH_1_0_0 = new RemoteVersion(1, 0, 0, "opensearch");
+    public static final RemoteVersion OPENSEARCH_2_0_0 = new RemoteVersion(2, 0, 0, "opensearch");
+    public static final RemoteVersion OPENSEARCH_3_1_0 = new RemoteVersion(3, 1, 0, "opensearch");
+
+    public RemoteVersion(int major, int minor, int revision, String distribution) {
+        this.major = major;
+        this.minor = minor;
+        this.revision = revision;
+        this.distribution = distribution != null ? distribution.toLowerCase(Locale.ROOT) : "elasticsearch";
```

Comment:
You could do an overloaded constructor, where not passing a distribution passes `"elasticsearch"` as the distribution. 

Alternatively, we could just make it a boolean -- like `isOpenSearch`.